### DISCUSSION TITLE
Add support for HarmonyOS hap files

### DISCUSSION THREAD

**Discussion by BUGProtectionAssociation** [breezy-weather/breezy-weather#1200] [Ideas]
### Describe your idea

I'm trying to port this program to the HarmonyOS NEXT system
But I'm not the root autor of this app object. I can't get the icp to publish it to offical store. The system requires me to provide a developer certificate.
You may be have try to do it if you have free time :)

### Other details

Just an idea, not mandatory
It's okay if you don't want to help

### Acknowledgements

- [X] I have searched the [existing issues](https://github.com/breezy-weather/breezy-weather/issues) and [existing discussions](https://github.com/breezy-weather/breezy-weather/discussions/categories/ideas) and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue or discussion.
- [X] The idea I’m suggesting is not in the [list of features that will not be implemented](https://github.com/breezy-weather/breezy-weather/blob/main/README.md#features-that-will-not-be-implemented).
- [X] I have updated the app to **[version 5.2.5 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**. Even if I’m getting updates from F-Droid default repo, I don’t need to uninstall/reinstall to update.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance). I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
I don't know HarmonyOS and its specifities, or who is the ICP, could you explain everything like I'm 5? Thanks!

**Reply by BUGProtectionAssociation** [reply]
There is some thing of [HarmonyOS](https://www.harmonyos.com/en/develop) may have some help

Just because of this
https://developer.huawei.com/consumer/en/doc/start/policycenter-0000001053129900#section16791121620317

**Reply by papjul** [reply]
I still don't understand, is that an Android system? If it is, what is preventing you from installing the app? Did they add some check before installing like MIUI does?

**Reply by BUGProtectionAssociation** [reply]
not Android，It's a new system
use ark instead of java

HarmonyOS 234 special Android（emui）
but HarmonyOS next not Android（new）
 so it can't run this app ，it need hap file to install

**Reply by papjul** [reply]
So HarmonyOS includes an Android emulator and you cannot run our app in this emulator without this hap file?

What is this hap file?

**Reply by BUGProtectionAssociation** [reply]
> hap file

It's a very difficult job. I may need to create a new version branch. If you are interested. You can install DevEco Studio to give it a try. Or just play around with this IDE and the hap program it compiles.

Thank you for your support, this may turn into a new project in the future

`GeometricWeather-->breezy-weather-->breezy-next-weather`

Branching of branches
Hahahaha

**Reply by papjul** [reply]
Why would you need a fork? If it is a different language/SDK, you need to create a different project. Please use a different name and icon than "Breezy Weather" in that case.

We don't compile on any IDE, we let GitHub do it. If you can provide the command lines to generate an hap file from an Android project, I can add hap files starting from next release.

https://github.com/breezy-weather/breezy-weather/blob/main/.github/workflows/push.yml
